### PR TITLE
feat(Host): RHICOMPL-3732 sort systems by compliance score

### PR DESCRIPTION
**Pull request by marleystipich2**
Tests are passing now 🎉  
Here's the SQL generated when sorting by score: 
```
Host Load (9.7ms)  SELECT "inventory"."hosts".* FROM "inventory"."hosts" LEFT OUTER JOIN "test_results" ON 
"test_results"."host_id" = "inventory"."hosts"."id" LEFT OUTER JOIN "profiles" ON "profiles"."id" = "test_results"."profile_id" 
WHERE "inventory"."hosts"."org_id" = $1 ORDER BY COALESCE("test_results"."score") ASC, "inventory"."hosts"."id" ASC 
LIMIT $2 OFFSET $3
```

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by marleystipich2** [comment]
Not sure how to test this or if it's possible because sorting like this doesn't work because `score` is nested in `testResultProfiles`:

```
query getSystems($policyId: ID) {
          systems(limit: 50, offset: 1, sortBy: ["score"]) {
              edges {
                  node {
                      id
                      testResultProfiles(policyId: $policyId) {
                          id
                          compliant
                          score
                          supported
                          ssgVersion
                      }
                  }
              }
          }
      }
```

**Comment by LightOfHeaven1994** [comment]
> Not sure how to test this or if it's possible.. Sorting like this doesn't work because `score` is nested in `testResultProfiles`:
> 
> ```
> query getSystems($policyId: ID) {
>           systems(limit: 50, offset: 1, sortBy: ["score"]) {
>               edges {
>                   node {
>                       id
>                       testResultProfiles(policyId: $policyId) {
>                           id
>                           compliant
>                           score
>                           supported
>                           ssgVersion
>                       }
>                   }
>               }
>           }
>       }
> ```

I'm just guessing here - maybe it's possible the same way as we sort systems by `ssgVersion`. As this attribute doesn't come from system itself but from profile (only when system reported so it is associated to some profile and we exactly know the version)

**Comment by marleystipich2** [comment]
@LightOfHeaven1994 I thought the same thing but there isn't a test for sorting by `ssgVersion` so that's why I wasn't able to go off of that ): I can try sorting by SSG version in the UI and get the GQL request from there and see if that helps though

File: test/graphql/queries/system_query_test.rb [hunk 1]
```diff
@@ -548,6 +548,70 @@ class SystemQueryTest < ActiveSupport::TestCase
       assert_equal @profile1.ssg_version,
                    returned_profiles.dig(0, 'ssgVersion')
     end
+
+    should 'properly sort by compliance score' do
```
Comment (marleystipich2) [inline test/graphql/queries/system_query_test.rb:552]: This test isn't working correctly right now even though the query should be correct. Still looking into why `sortBy` isn't working correctly when sorting anything nested under `testResultProfiles`.

File: app/models/host.rb [hunk 2]
```diff
@@ -78,6 +82,10 @@ class Host < ApplicationRecord
     )
   }
 
+  scope :joins_test_results, lambda {
```
Comment (skateman) [inline app/models/host.rb:85]: Semantically this is incorrect, should be:
```suggestion
  scope :joins_test_result_profiles, lambda {
```

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Pull request merged**